### PR TITLE
feat: add more metrics to investigate tx broadcast issue

### PR DESCRIPTION
**Pull request by 0xmountaintop**
part 2 of https://github.com/scroll-tech/go-ethereum/pull/1005

### CODE REVIEW

**Review by Thegaram — DISMISSED** [review]

**Review by Thegaram — APPROVED** [review]

File: eth/protocols/eth/broadcast.go [hunk 1]
```diff
@@ -22,6 +22,18 @@ import (
 	"github.com/scroll-tech/go-ethereum/common"
 	"github.com/scroll-tech/go-ethereum/core/types"
 	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/metrics"
+)
+
+var (
+	broadcastSendTxsLenGauge    = metrics.NewRegisteredGauge("eth/protocols/eth/broadcast/direct/txs", nil)
+	broadcastSendTxsFailGauge   = metrics.NewRegisteredGauge("eth/protocols/eth/broadcast/direct/fail", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/broadcast.go:30]: change to counter

File: eth/protocols/eth/broadcast.go [hunk 2]
```diff
@@ -22,6 +22,18 @@ import (
...
+	broadcastSendHashesLenGauge = metrics.NewRegisteredGauge("eth/protocols/eth/broadcast/direct/hashes", nil)
+	broadcastSendQueueLenGauge  = metrics.NewRegisteredGauge("eth/protocols/eth/broadcast/direct/queue", nil)
+	broadcastAnnoTxsLenGauge    = metrics.NewRegisteredGauge("eth/protocols/eth/broadcast/anno/txs", nil)
+	broadcastAnnoTxsFailGauge   = metrics.NewRegisteredGauge("eth/protocols/eth/broadcast/anno/fail", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/broadcast.go:34]: change to counter

File: eth/protocols/eth/handlers.go [hunk 3]
```diff
@@ -23,10 +23,25 @@ import (
 	"github.com/scroll-tech/go-ethereum/common"
 	"github.com/scroll-tech/go-ethereum/core/types"
 	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/metrics"
 	"github.com/scroll-tech/go-ethereum/rlp"
 	"github.com/scroll-tech/go-ethereum/trie"
 )
 
+var (
+	newPooledTxHashesFailGauge    = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/newpooledtxhashes/fail", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/handlers.go:32]: change to counter

File: eth/protocols/eth/handlers.go [hunk 4]
```diff
@@ -23,10 +23,25 @@ import (
...
+	newPooledTxHashesLenGauge     = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/newpooledtxhashes/len", nil)
+	getPooledTxsFailGauge         = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/getpooledtxs/fail", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/handlers.go:34]: change to counter

File: eth/protocols/eth/handlers.go [hunk 5]
```diff
@@ -23,10 +23,25 @@ import (
...
+	getPooledTxsQueryLenGauge     = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/getpooledtxs/query", nil)
+	getPooledTxsRetrievedLenGauge = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/getpooledtxs/retrieved", nil)
+	handleTxsFailGauge            = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/handletxs/fail", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/handlers.go:37]: change to counter

File: eth/protocols/eth/handlers.go [hunk 6]
```diff
@@ -23,10 +23,25 @@ import (
...
+	handleTxsLenGauge             = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/handletxs/len", nil)
+	handleTxsNilGauge             = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/handletxs/nil", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/handlers.go:39]: change to counter

File: eth/protocols/eth/handlers.go [hunk 7]
```diff
@@ -23,10 +23,25 @@ import (
...
+	pooledTxs66FailGauge          = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/pooledtxs66/fail", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/handlers.go:40]: change to counter

File: eth/protocols/eth/handlers.go [hunk 8]
```diff
@@ -23,10 +23,25 @@ import (
...
+	pooledTxs66LenGauge           = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/pooledtxs66/len", nil)
+	pooledTxs66NillGauge          = metrics.NewRegisteredGauge("eth/protocols/eth/handlers/pooledtxs66/nil", nil)
```
Comment (0xmountaintop) [inline eth/protocols/eth/handlers.go:42]: change to counter

**Pull request merged**